app/controllers/datasets_controller.rb

```diff
@@ -105,6 +105,29 @@ def publish
     end
   end
 
+  def confirm_delete
+    @dataset = current_dataset
+    if @dataset.published?
+      @dataset.errors.add(:delete_prevent, 'Published datasets cannot be deleted')
+    else
+      flash[:confirm_delete] = 'Are you sure you want to delete this dataset?'
```

Comment:
You could use `flash[:alert]` or something of that ilk, i believe rails has a default for it.